test/tests.hs

```diff
@@ -75,14 +74,18 @@ main = defaultMain $ testGroup "discrimination"
     , testGrouping (Proxy :: Proxy (Word8,Word8))
     , testGrouping (Proxy :: Proxy (Word8,Word8,Word8))
     , testGrouping (Proxy :: Proxy (Word8,Word8,Word8,Word8))
+    , testGrouping (Proxy :: Proxy Rational)
     , testGrouping (Proxy :: Proxy (Complex Word8))
     , testGrouping (Proxy :: Proxy (Maybe Word8))
     , testGrouping (Proxy :: Proxy (Either Word8 Word8))
     , testGrouping (Proxy :: Proxy Char)
     , testGrouping (Proxy :: Proxy String)
     , testGrouping (Proxy :: Proxy (NonEmpty Int))
-    -- , testGrouping (Proxy :: Proxy Natural) -- broken!
-    -- , testGrouping (Proxy :: Proxy Integer) -- broken!
+    , testGrouping (Proxy :: Proxy Natural)
+    , testGrouping (Proxy :: Proxy Integer)
+
+    , testGrouping' (\xs -> L.foldl' (*) 1 (map fromIntegral (xs :: [Word64])) :: Natural)
+    , testGrouping' (\xs -> L.foldl' (*) 1 (map fromIntegral (xs :: [Int64]))  :: Integer)
```

Comment:
Just another curiosity question. Where these added mostly to track down the reverse limb issue, or was there some other issue that  the standard `Natural` and `Integer` ones weren't  triggering?

Reply:
Correct guess. Ordinary `Arbitrary Integer` instance doesn't generate huge (in absolute value) numbers.

Reply:
:+1: